src/Reflection/FunctionReflectionFactory.php

```diff
@@ -29,6 +30,7 @@ public function create(
 		Assertions $asserts,
 		?string $phpDocComment,
 		array $phpDocParameterOutTypes,
+		TrinaryLogic $returnsByReference,
```

Comment:
This doesn't have to be part of the constructor, you can just read it from ReflectionFunction.